src/bin/fxrunner/main.rs

```diff
@@ -0,0 +1,50 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+mod config;
+
+use std::path::PathBuf;
+use std::process::exit;
+
+use slog::{error, info, Logger};
+use snafu::{ResultExt, Snafu};
+use structopt::StructOpt;
+
+use crate::config::Config;
+use fxrecord::{build_logger, read_config, ConfigError};
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "fxrunner", about = "Start FxRunner")]
+struct Options {
+    /// The configuration file to use.
+    #[structopt(long = "config", default_value = "fxrecord.toml")]
+    config_path: PathBuf,
+}
+
+fn main() {
+    let options = Options::from_args();
+    let log = build_logger();
+
+    if let Err(e) = fxrunner(log.clone(), options) {
+        error!(log, "unexpected error"; "error" => %e);
```

Comment:
This is a little incongruous for the end user because the output has a very specific error message (parsing error, I/O error, etc) and then it's prefixed by 'unexpected error'. Perhaps you could just print the error and not the prefix?

Same would apply to fxrecorder, too.

Reply:
the unexpected error bit is because these errors should not happen, i.e., these are the exceptional cases if they've bubbled up all the way up to main.

Most errors that occur won't make it this far.

Reply:
I think that they will. I ran the tool and saw this very output in an unexceptional, exceptional situation: a field in the input that is not supported by the containing section.

Reply:
Also part of this is we are using structured logging, where we can set Key-Value pairs. We might end up switching to a logging format that, eg uses JSON and then the error format would look like

```json
{
     "event": "unexpected error",
     "error": "could not open config file"
}
```

In this case, if we put the error message in the event field, we wouldn't be able to grep through the logs for all errors since there would be no error key.

Also, not being able to parse the config file is exceptional. If we cant interpret it, then we cant run.

Reply:
All I am saying is that if we have code that handles the error and formats it, then it's not unexpected. I see unexpected as an error that the code was not designed to handle. So, something like this:

```
Oct 07 17:29:16.818 ERRO unexpected error, error: Could not parse config file `fxrecord.toml': missing field `host`
```

does not make sense. We have enough "intelligence" in the code to format the error properly, and report it succinctly and informatively. It shouldn't be labeled unexpected. 

Yes, I am bikeshedding. Ultimately it's your choice.

Reply:
These logs are not intended for human consumption. People aren't going to be running this command. Its going to be triggered by try jobs and the results will be reported there.